relay-filter/src/config.rs

```diff
@@ -192,6 +196,86 @@ impl LegacyBrowsersFilterConfig {
     }
 }
 
+/// Configuration for a generic filter.
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GenericFilterConfig {
+    /// Specifies whether this filter is enabled.
+    pub is_enabled: bool,
+    /// The condition for the filter.
+    pub condition: Option<RuleCondition>,
+}
+
+impl GenericFilterConfig {
+    /// Returns true if the filter is not enabled or the condition is not supported.
+    pub fn is_empty(&self) -> bool {
+        !self.is_enabled || matches!(self.condition, Some(RuleCondition::Unsupported))
```

Comment:
What should the behavior be if there's no condition, always filter?

Reply:
Here the semantics need to be extended. I would classify as filter as disabled iff (the condition is OR):
* is_enabled = False
* condition is None
* condition is Unsupported

Reply:
But since the `Unsupported` handling is done directly in the `matches` of `RuleCondition` and now we switched to an optional, I will just keep `is_enabled = False or condition is None`.